include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,617 @@
+                            enthalpy_coverages(enthalpy_coverages_),
+                            enthalpies(enthalpies_),
+                            entropy_coverages(entropy_coverages_),
+                            entropies(entropies_) {}
+    InterpolativeDependency() {}
+    std::string name_k; //!< name of species which thermodynamics are calculated for [integer]
+    std::string name_j; //!< name of species which thermodynamics are dependent on [integer]
+    vector_fp enthalpy_coverages; /*!< array of coverages for coverage-dependent 
+                                   *   enthalpy interpolation [dimensionless]
+                                   */
+    vector_fp enthalpies; /*!< array of enthalpies at corresponding coverages 
+                           *   in enthalpy-coverages [J/kmol]
+                           */
+    vector_fp entropy_coverages; /*!< array of coverages for coverage-dependent
+                                  *   entropy interpolation [dimensionless]
+                                  */
+    vector_fp entropies; /*!< array of entropies at corresponding coverages 
+                          *   in entropy-coverages [J/kmol/K]
+                          */
+};
+
+//! Set of parameters modifying SurfPhase heat capacity based on fractional surface coverages 
+//! using a quadratic model.
+struct HeatCapacityDependency
+{
+    //! Constructor
+    //! @param name_k_  name of species which thermodynamics are calculated for [integer]
+    //! @param name_j_  name of species which thermodynamics are dependent on [integer]
+    //! @param cpcov_a_  log model coefficient a [J/kmol/K]
+    //! @param cpcov_b_  log model coefficient b [J/kmol/K]
+
+    HeatCapacityDependency(std::string name_k_, std::string name_j_,
+                           double cpcov_a_, double cpcov_b_):
+                           name_k(name_k_), name_j(name_j_),
+                           cpcov_a(cpcov_a_), cpcov_b(cpcov_b_) {}
+    HeatCapacityDependency() {}
+    std::string name_k; //!< name of species which thermodynamics are calculated for [integer]
+    std::string name_j; //!< name of species which thermodynamics are dependent on [integer]
+    double cpcov_a; //!< log model coefficient a [J/kmol/K]
+    double cpcov_b; //!< log model coefficient b [J/kmol/K]
+};
+
+class CoverageDependentSurfPhase : public SurfPhase
+{
+public:
+    //! Constructor.
+    /*!
+     *  @param n0 Site Density of the Surface Phase
+     *            Units: kmol m^-2.
+     */
+    CoverageDependentSurfPhase(doublereal n0 = 1.0);
+
+    //! Construct and initialize a CoverageDependentSurfPhase ThermoPhase object directly 
+    //! from an ASCII input file
+    /*!
+     * @param infile name of the input file
+     * @param id     name of the phase id in the file
+     *               If this is blank, the first phase in the file is used
+     */
+    explicit CoverageDependentSurfPhase(const std::string& infile, const std::string& id);
```

Comment:
The phase name default to the empty string
```suggestion
    explicit CoverageDependentSurfPhase(const std::string& infile, const std::string& id="");
```